parlai/tasks/self_chat/worlds.py

```diff
@@ -4,52 +4,139 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+import copy
 import random
+from typing import Any, Dict, List, Optional
 
-from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
+from parlai.core.agents import Agent
+from parlai.core.worlds import create_task, DialogPartnerWorld, validate
 
 
-class InteractiveWorld(DialogPartnerWorld):
+def load_openers(opt) -> Optional[List[str]]:
+    base_task = opt['task'].split(':')[0]
+    if base_task == 'self_chat':
+        # TODO(#2284): Load default openers from s3
+        return None
+
+    print('[ loading conversation openers... ]')
+    # create dummy task so we can get openers from the data
+    task_opt = copy.deepcopy(opt)
+    task_opt['task'] = base_task
+
+    # default train will loop forever, but evalmode will stop after one epoch
+    datatype = task_opt['datatype']
+    if 'train' in datatype and 'evalmode' not in datatype:
+        task_opt['datatype'] = f'{datatype}:evalmode'
+    task_opt['interactive_task'] = False
+    task_agent = RepeatLabelAgent(task_opt)
+    task_world = create_task(task_opt, task_agent)
+
+    # run through task data, collecting all first messages
+    openers = set()
+    is_first_turn = True
+    while not task_world.epoch_done():
+        task_world.parley()
+        msg = task_world.get_acts()[0]
+        # add only the first message in the episode
+        if is_first_turn and msg.get('text'):
+            openers.add(msg.get('text'))
```

Comment:
if the 'text' field does not exist in the msg then we probably should not add None to the set of openers

Reply:
That's why we check `msg.get('text')` on the line above :)

Reply:
In that case why don't we just add `msg['text']`?

Reply:
That's fair! I suppose if someone accidentally removes the check above, we want this to throw an exception instead of silently adding None to the set.